packages/frontend/src/lib/pnl.ts

```diff
@@ -1,21 +1,93 @@
 import BigNumber from 'bignumber.js'
 
+import { VaultHistory_vaultHistories } from '../queries/squeeth/__generated__/VaultHistory'
+import { toTokenAmount } from '@utils/calculations'
+import { Action } from '@constants/enums'
+
 type ShortPnLParams = {
   wethAmount: BigNumber
   buyQuote: BigNumber
   ethPrice: BigNumber
+  ethCollateralPnl: BigNumber
 }
 
-export function calcUnrealizedPnl({ wethAmount, buyQuote, ethPrice }: ShortPnLParams) {
+export function calcUnrealizedPnl({ wethAmount, buyQuote, ethPrice, ethCollateralPnl }: ShortPnLParams) {
   if (
     wethAmount.isEqualTo(0) ||
     !wethAmount.isFinite() ||
     buyQuote.isEqualTo(0) ||
     !buyQuote.isFinite() ||
     ethPrice.isEqualTo(0) ||
-    !ethPrice.isFinite()
+    !ethPrice.isFinite() ||
+    ethCollateralPnl.isEqualTo(0) ||
+    !ethCollateralPnl.isFinite()
   ) {
     return new BigNumber(0)
   }
-  return wethAmount.minus(buyQuote).multipliedBy(ethPrice)
+
+  return wethAmount.minus(buyQuote).multipliedBy(ethPrice).plus(ethCollateralPnl)
```

Comment:
```suggestion
  return buyQuote.minus(wethAmount).multipliedBy(ethPrice).plus(ethCollateralPnl)
```

Reply:
This was the part of the formula that @KMKoushik corrected the other day. That's why I changed it. See https://opyn.slack.com/archives/C02R5NAT6BA/p1642492674016200